json-fleece-core/src/Fleece/Core/Schemas.hs

```diff
@@ -436,6 +437,20 @@ zonedTime :: Fleece schema => schema Time.ZonedTime
 zonedTime =
   iso8601Formatted "ZonedTime" ISO8601.iso8601Format AttoTime.zonedTime
 
+dayWithFormat :: Fleece schema => String -> schema Time.Day
+dayWithFormat formatString =
+  let
+    decode raw =
+      case Time.parseTimeM False Time.defaultTimeLocale formatString raw of
+        Just success -> Right success
+        Nothing -> Left "Invalid date in custom format"
```

Comment:
Maybe we should give the format in the error message here?

Reply:
Added in 93bc26091eb7d665f01d05e84f1774e0c38a6fe6